### PR TITLE
Heitor's updates to publish 0.1 - part 1

### PR DESCRIPTION
**Pull request by hguerra**

### CODE REVIEW

File: lua/Layout.lua [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	if not belong(data.base, {"roadmap", "satellite", "hybrid", "terrain"}) then
+		customError("Basemap '"..data.base.."' is not supported.")
+	end
+
+	verifyRange(data.zoom, 0, 20, "Argument 'zoom' must be a number >= 0 and <= 20, got '"..data.zoom.."'.")
+	verifyRange(data.minZoom, 0, 20, "Argument 'minZoom' must be a number >= 0 and <= 20, got '"..data.minZoom.."'.")
+	verifyRange(data.maxZoom, 0, 20, "Argument 'maxZoom' must be a number >= 0 and <= 20, got '"..data.maxZoom.."'.")
+
+	if data.minZoom > data.maxZoom then
+		customError("Argument 'minZoom' should be less than 'maxZoom' ("..data.maxZoom..").")
```
Comment (pedro-andrade-inpe) [inline lua/Layout.lua]: @As you added the value of maxZoom here, add also the value of minZoom.

File: lua/Layout.lua [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	end
+
+	verifyRange(data.center.lat, -90, 90, "Center 'lat' must be a number >= -90 and <= 90, got '"..data.center.lat.."'.")
+	verifyRange(data.center.long, -180, 180, "Center 'long' must be a number >= -180 and <= 180, got '"..data.center.long.."'.")
+
```
Comment (pedro-andrade-inpe) [inline lua/Layout.lua]: @check unnecessary arguments in `data.center`.

Comment (pedro-andrade-inpe) [inline lua/Layout.lua]: It must also be a named table.

File: lua/Application.lua [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+	postgis = "POSTGIS",
+	access = "ADO"
+}
+
+local Templates = {}
+local templateDir = Directory(packageInfo("publish").path.."/lua/layout")
+local function registerApplicationTemplate(data)
+	mandatoryTableArgument(data, "input", "string")
+	mandatoryTableArgument(data, "output", "string")
+	verify(isTable(data.model), incompatibleTypeMsg("model", "table", data.model))
```
Comment (hguerra) [inline lua/Application.lua]: I used that to set Layout as a template for `publish.js`. I will review this.

**Comment by pedro-andrade-inpe** [comment]
This pull request also closes #13.

**Pull request merged**